examples/astro-stores-demo/src/components/media/MediaGallery.tsx

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import { MediaGallery as MediaGalleryRadix, WixMediaImage } from "@wix/headless-media/react";
+
+export const Root: React.FC<React.ComponentProps<typeof MediaGalleryRadix.Root>> = (props) => (
+  <MediaGalleryRadix.Root {...props} />
+);
+
+export const Viewport: React.FC = () => (
+  <MediaGalleryRadix.Viewport>
+    {({ src, alt }: { src: string | null; alt: string }) => (
```

Comment:
this all logic and wiring should be in the headless component and not in the user implementation, also states should be based on `data-[]` props and not injected props with `if`